sdk/deviceprovisioningservices/azure-iot-deviceprovisioningservice/azure/iot/deviceprovisioningservice/_generated/operations/_operations.py

```diff
@@ -0,0 +1,6904 @@
+            request, stream=_stream, **kwargs
+        )
+
+        response = pipeline_response.http_response
+
+        if response.status_code not in [200]:
+            map_error(status_code=response.status_code, response=response, error_map=error_map)
+            raise HttpResponseError(response=response)
+
+        if response.content:
+            deserialized = response.json()
+        else:
+            deserialized = None
+
+        if cls:
+            return cls(pipeline_response, cast(JSON, deserialized), {})
+
+        return cast(JSON, deserialized)
+
+    @distributed_trace
+    def delete(  # pylint: disable=inconsistent-return-statements
+        self, id: str, *, if_match: Optional[str] = None, **kwargs: Any
+    ) -> None:
+        """Delete a device enrollment record.
+
+        Delete a device enrollment record.
+
+        :param id: This id is used to uniquely identify a device registration of an enrollment. A
+         case-insensitive string (up to 128 characters long) of alphanumeric characters plus certain
+         special characters : . _ -. No special characters allowed at start or end. Required.
+        :type id: str
+        :keyword if_match: The ETag of the enrollment record. Default value is None.
+        :paramtype if_match: str
+        :return: None
+        :rtype: None
+        :raises ~azure.core.exceptions.HttpResponseError:
+        """
+        error_map = {
+            401: ClientAuthenticationError,
+            404: ResourceNotFoundError,
+            409: ResourceExistsError,
+            304: ResourceNotModifiedError,
+        }
+        error_map.update(kwargs.pop("error_map", {}) or {})
+
+        _headers = kwargs.pop("headers", {}) or {}
+        _params = kwargs.pop("params", {}) or {}
+
+        cls: ClsType[None] = kwargs.pop("cls", None)
+
+        request = build_individual_enrollment_delete_request(
+            id=id,
+            if_match=if_match,
+            api_version=self._config.api_version,
+            headers=_headers,
+            params=_params,
+        )
+        request.url = self._client.format_url(request.url)
+
+        _stream = False
+        pipeline_response: PipelineResponse = self._client._pipeline.run(  # pylint: disable=protected-access
+            request, stream=_stream, **kwargs
+        )
+
+        response = pipeline_response.http_response
+
+        if response.status_code not in [204]:
+            map_error(status_code=response.status_code, response=response, error_map=error_map)
+            raise HttpResponseError(response=response)
+
+        if cls:
+            return cls(pipeline_response, None, {})
+
+    @overload
+    def query(
+        self,
+        query_specification: JSON,
+        *,
+        x_ms_max_item_count: Optional[int] = None,
+        x_ms_continuation: Optional[str] = None,
+        content_type: str = "application/json",
+        **kwargs: Any
+    ) -> List[JSON]:
```

Comment:
We talked about supporting azure.core.ItemPaged[JSON] as return type, this should hide x_ms_max_items_count and x_ms_continuation

Reply:
Could be opportunity to leverage the patch pattern against the generated client.